### PR TITLE
F4 uart derivation interrupt fixes

### PR DESCRIPTION
**Pull request by burrbull**
retry of #754 

cc @tim-seoss

### CODE REVIEW

**Comment by burrbull** [comment]
Looks like this fixes all issues. Merge, please.

**Comment by tim-seoss** [comment]
LGTM, thanks for picking this up (was hoping to get back to it next month).

File: devices/stm32f405.yaml [hunk 1]
```diff
@@ -63,6 +63,26 @@ MPU:
...
+UART8:
+  _add:
+    _interrupts:
+      UART8:
+        description: USART 8 global interrupt
+        value: 83
+  _delete:
+    _interrupts:
+      - UART5
+
```
Comment (adamgreig) [inline devices/stm32f405.yaml:85]: STM32F405 doesn't have UART7 or UART8, only 1-6.

**Review by adamgreig — COMMENTED** [review]
STM32F405 doesn't have UART7/8 at all, including its interrupts - the reference manual includes F429/439 which do have those, there's two separate tables for interrupts.

Everything else LGTM.

**Review by adamgreig — APPROVED** [review]
thanks!

bors r+

**Pull request merged**